nntrainer/layers/preprocess_flip_layer.cpp

```diff
@@ -34,53 +33,16 @@ void PreprocessFlipLayer::finalize(InitLayerContext &context) {
 }
 
 void PreprocessFlipLayer::setProperty(const std::vector<std::string> &values) {
-  /// @todo: deprecate this in favor of loadProperties
-  for (unsigned int i = 0; i < values.size(); ++i) {
-    std::string key;
-    std::string value;
-    std::stringstream ss;
-
-    if (getKeyValue(values[i], key, value) != ML_ERROR_NONE) {
-      throw std::invalid_argument("Error parsing the property: " + values[i]);
-    }
-
-    if (value.empty()) {
-      ss << "value is empty: key: " << key << ", value: " << value;
-      throw std::invalid_argument(ss.str());
-    }
-
-    /// @note this calls derived setProperty if available
-    setProperty(key, value);
-  }
-}
-
-void PreprocessFlipLayer::setProperty(const std::string &type_str,
-                                      const std::string &value) {
-  using PropertyType = nntrainer::Layer::PropertyType;
-  nntrainer::Layer::PropertyType type =
-    static_cast<nntrainer::Layer::PropertyType>(parseLayerProperty(type_str));
-
-  switch (type) {
-  case PropertyType::flip_direction: {
-    if (istrequal(value, flip_horizontal)) {
-      flipdirection = FlipDirection::horizontal;
-    } else if (istrequal(value, flip_vertical)) {
-      flipdirection = FlipDirection::vertical;
-    } else if (istrequal(value, flip_horizontal_vertical)) {
-      flipdirection = FlipDirection::horizontal_and_vertical;
-    } else {
-      throw std::invalid_argument("Argument flip direction is invalid");
-    }
-  } break;
-  default:
-    std::string msg =
-      "[PreprocessFlipLayer] Unknown Layer Property Key for value " +
-      std::string(value);
-    throw exception::not_supported(msg);
-  }
+  auto remain_props = loadProperties(values, preprocess_flip_props);
+  NNTR_THROW_IF(!remain_props.empty(), std::invalid_argument)
```

Comment:
You can consider (now or in a layer PR) providing a default implementation to LayerDevel which throws. It will simplify each layer to additionally throw and check for remaining props, and removes the burden from layers which do not want any property to perform this check anyway.